### PR TITLE
new module to manipulate DAG table

### PR DESCRIPTION
**Pull request by ivanbojer**
##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
panos_dag

##### ANSIBLE VERSION

```
ansible 2.1.0.0 
config file = 
configured module search path = Default w/o overrides
```

##### SUMMARY
New panos module that allows for Dynamic Address Group generation. DAG's are dynamic run-time objects that are used in FW policies. The typical example would be a white-list of IP's that is updated every 5 minutes.

### CODE REVIEW

File: lib/ansible/modules/network/panos/panos_dag.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    type: string
+'''
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+from ansible.module_utils.basic import AnsibleModule
+
+try:
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_dag.py:84]: shared module?

File: lib/ansible/modules/network/panos/panos_dag.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+EXAMPLES = '''
+- name: dag
+  panos_dag:
+    ip_address: "192.168.1.1"
+    password: "admin"
+    dag_name: "dag-1"
+    dag_filter: "'aws-tag.aws:cloudformation:logical-id.ServerInstance' and 'instanceState.running'"
+'''
+
+RETURN = '''
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_dag.py]: return value does not match RETURN doc string

File: lib/ansible/modules/network/panos/panos_dag.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        dag_name=dict(required=True),
+        dag_filter=dict(required=True),
+        commit=dict(type='bool', default=True)
+    )
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=False)
+    if not HAS_LIB:
+        module.fail_json(msg='pan-python is required for this module')
+
+    ip_address = module.params["ip_address"]
+    if not ip_address:
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_dag.py]: not needed

File: lib/ansible/modules/network/panos/panos_dag.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        module.fail_json(msg="ip_address should be specified")
+    password = module.params["password"]
+    if not password:
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_dag.py]: not needed

File: lib/ansible/modules/network/panos/panos_dag.py [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+        hostname=ip_address,
+        api_username=username,
+        api_password=password
+    )
+
+    dag_name = module.params['dag_name']
+    if not dag_name:
+        module.fail_json(msg="dag_name is required")
+    dag_filter = module.params['dag_filter']
+    if not dag_filter:
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_dag.py]: not needed

File: lib/ansible/modules/network/panos/panos_dag.py [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+    username = module.params['username']
+
+    xapi = pan.xapi.PanXapi(
+        hostname=ip_address,
+        api_username=username,
+        api_password=password
+    )
+
+    dag_name = module.params['dag_name']
+    if not dag_name:
```
Comment (privateip) [inline lib/ansible/modules/network/panos/panos_dag.py]: not needed

**Review by privateip — CHANGES_REQUESTED** [review]
consider adding unit tests

File: lib/ansible/modules/network/panos/panos_dag.py (hunk 2 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_dag.py]: no return so I copied default return string from wakeonlan.py

File: lib/ansible/modules/network/panos/panos_dag.py (hunk 1 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_dag.py:84]: no

File: lib/ansible/modules/network/panos/panos_dag.py (hunk 3 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_dag.py]: ok

File: lib/ansible/modules/network/panos/panos_dag.py (hunk 4 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_dag.py]: ok

File: lib/ansible/modules/network/panos/panos_dag.py (hunk 6 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_dag.py]: ok

File: lib/ansible/modules/network/panos/panos_dag.py (hunk 5 above)
Comment (ivanbojer) [reply lib/ansible/modules/network/panos/panos_dag.py]: ok

**Comment by ivanbojer** [comment]
Thank you for the review. I've incorporated all the changes requested.

**Comment by ivanbojer** [comment]
ready_for_review

**Pull request merged**